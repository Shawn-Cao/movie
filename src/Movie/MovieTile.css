.MovieTile {
  position: absolute;
}

.poster-img {
  position: absolute;
  left: 0;
  top: 0;
  width: 154px;
  height: 278px;
  border: 1px solid black;
  transition-property: left, top, height, width;
  transition-duration: 1s;
}

.poster-img:hover {
  left: -25px;
  top: -14px;
  width: 200px;
  height: 361px;
  z-index: 100;
}
