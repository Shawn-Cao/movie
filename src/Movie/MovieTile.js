import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './MovieTile.css';

class MovieTile extends Component {
  static get propTypes() {
    return {
      movie: PropTypes.object.isRequired
    };
  }

  render() {
    return (
      <div className="MovieTile">
        <img src={`http://image.tmdb.org/t/p/w185${this.props.movie.poster_path}`}
          className="poster-img" alt={`poster of movie: ${this.props.movie.title}`} />
      </div>
    );
  }
}

export default MovieTile;
