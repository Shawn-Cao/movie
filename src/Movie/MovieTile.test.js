import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import MovieTile from './MovieTile';


const movie = {
  "vote_count": 1724,
  "id": 297762,
  "video": false,
  "vote_average": 7,
  "title": "Wonder Woman",
  "popularity": 99.147603,
  "poster_path": "/gfJGlDaHuWimErCr5Ql0I8x9QSy.jpg",
  "original_language": "en",
  "original_title": "Wonder Woman",
  "genre_ids": [
    28,
    12,
    14,
    878
  ],
  "backdrop_path": "/hA5oCgvgCxj5MEWcLpjXXTwEANF.jpg",
  "adult": false,
  "overview": "An Amazon princess comes to the world of Man to become the greatest of the female superheroes.",
  "release_date": "2017-05-30"
};

describe('MovieTime', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MovieTile movie={movie} />, div);
  });
  it('renders matching snapshot', () =>{
    const tree = renderer.create(
      <MovieTile movie={movie} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
