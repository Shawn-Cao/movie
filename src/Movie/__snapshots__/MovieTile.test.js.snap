// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTime renders matching snapshot 1`] = `
<div
  className="MovieTile"
>
  <img
    alt="poster of movie: Wonder Woman"
    className="poster-img"
    src="http://image.tmdb.org/t/p/w185/gfJGlDaHuWimErCr5Ql0I8x9QSy.jpg"
  />
</div>
`;
